File: week-2-ruby-exercices/acronym/acronym.rb
#
# def makeAcronym(sentence)
#     if sentence.is_a?(String)
#         # sentence.split.map do |word|
#         #     word[0].upcase
#         # end.join
#         array = sentence.split.map do |word|
#             word[0].upcase
#         end
#         array.join
#     else
#         "Not a String"
#     end
#
#     # if condition
#     #
#     # end
# end

def makeAcronym(sentence)
    return "Not a string" unless sentence.is_a?(String)
    array = sentence.split.map do |word|
        word[0].upcase
    end
    array.join
end

# def makeAcronym(sentence)
#     return "Not a string" unless sentence.is_a?(String)
#     sentence.split.map do |word|
#         word[0].upcase
#     end.join
# end

p makeAcronym("kslks jfyj")
p makeAcronym(3)
p makeAcronym("Hello codewarrior")
p makeAcronym("a42")
p makeAcronym("42")
p makeAcronym([2,12])

# 'Hello codewarrior' -> 'HC'
#
# 'a42' -> 'Not letters'
#
# 42 -> 'Not a string'
#
# [2,12] -> 'Not a string'
#
# {name: 'Abraham'} -> 'Not a string'

File: week-8-js-exercises/acronym/acronym.js
function makeAcronym(value) {

  if (typeof value !== 'string') {
    return "Not a string";
  }
  if (value.length !== 0 && value.match(/^[a-zA-Z ]+$/) === null ) {
    return "Not letters";
  }
  else {
    return value.split(" ").map(function(word) {
      return word.charAt(0).toUpperCase();
    }).join("");
  }
}

console.log(makeAcronym("Hello World"));
console.log(makeAcronym("Hello codewarrior"));
console.log(makeAcronym("a42"));
console.log(makeAcronym("42"));
console.log(makeAcronym([2,12]));
